### PR TITLE
Return notes from ArchivesSpaceClient and allow them to be edited

### PR DESCRIPTION
**Pull request by mistydemeo**
This currently supports a subset of ArchivesSpace's notes functionality, supporting a single block of text in a single note. (ArchivesSpace allows arbitrary numbers of notes, and each note may contain arbitrary numbers of blocks of text.)

### CODE REVIEW

**Comment by mistydemeo** [comment]
Updated to allow start and end dates, and date expressions, to be edited.

File: src/archivematicaCommon/lib/archivesspace/client.py [hunk 1]
```diff
@@ -118,6 +118,26 @@ def _delete(self, url, params={}, expected_response=200):
                              params=params,
                              expected_response=expected_response)
 
+    def _format_notes(self, record):
+        """
+        Extracts notes from a record and reformats them in a simplified format.
+        """
+        if len(record['notes']) == 0:
```
Comment (Hwesta) [inline src/archivematicaCommon/lib/archivesspace/client.py]: Are notes guaranteed to exist? Should this be `.get('notes', [])`? Can we check for truthiness?

File: src/archivematicaCommon/lib/archivesspace/client.py [hunk 2]
```diff
@@ -165,6 +189,47 @@ def edit_record(self, new_record):
...
+            # If the record already has at least one note, then replace the first note
+            # within that record with this one.
+            if len(record['notes']) == 0:
+                record['notes'] = [new_note]
+            else:
+                record['notes'][0] = new_note
+
+            fields_updated = True
+
+        if 'start_date' in new_record:
```
Comment (Hwesta) [inline src/archivematicaCommon/lib/archivesspace/client.py:208]: end_date and date_expression are meaningless if there's no start_date?

Comment (mistydemeo) [inline src/archivematicaCommon/lib/archivesspace/client.py:208]: That's right. `start_date` is mandatory.

File: src/archivematicaCommon/lib/archivesspace/client.py (hunk 1 above)
Comment (mistydemeo) [inline src/archivematicaCommon/lib/archivesspace/client.py]: Records are guaranteed to contain a `notes` property. Could check truthiness though.

**Comment by Hwesta** [comment]
Tests for new functionality?

:+1:

**Comment by mistydemeo** [comment]
Added a test!

**Comment by Hwesta** [comment]
:+1:

**Pull request merged**